presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/Util.java

```diff
@@ -87,4 +91,35 @@ public static PlanNode transpose(PlanNode parent, PlanNode child)
                         node,
                         Assignments.identity(restrictedOutputs)));
     }
+
+    public static List<Set<Symbol>> requiredJoinInputs(JoinNode node, Set<Symbol> requiredJoinOutputs)
+    {
+        Set<Symbol> globallyUsable = requiredJoinOutputs;
```

Comment:
I did it this way because below the if block, I wanted to use globallyUsable twice, once for the left child and once for the right.  If I just copy-paste the common code and make two functions, one for the left child and one for the right, then a single builder would work.